src/opera/pge/disp_s1/disp_s1_pge.py

```diff
@@ -589,6 +593,33 @@ def _qa_log_filename(self):
         """
         return self._ancillary_filename() + ".qa.log"
 
+    def augment_measured_parameters(self, measured_parameters):
+        descriptions_file = self.runconfig.iso_measured_parameter_descriptions
+
+        new_measured_parameters = {}
+
+        if descriptions_file is not None:
+            with open(descriptions_file) as f:
+                descriptions = yaml.safe_load(f)
+        else:
+            msg = ('Measured parameters configuration is needed to extract the measured parameters attributes from the'
+                   'DISP-S1 metadata')
+            self.logger.critical(self.name, ErrorCode.ISO_METADATA_DESCRIPTIONS_CONFIG_NOT_FOUND, msg)
+
+        for parameter_var_name in descriptions:
+            key_path = parameter_var_name.split(MEASURED_PARAMETER_PATH_SEPARATOR)
+
+            mp = measured_parameters
+
+            while len(key_path) > 0:
+                mp = mp[key_path.pop(0)]
```

Comment:
This could use a try/except block that catches `KeyError` and logs a descriptive error containing the full `parameter_var_name`. Will be helpful to catch typos in the description file down the line.

Reply:
Did this here and a couple of these other suggestions to CSLC as well as the code is identical there